3lvia/kunde-elvia-web
# Fix/omega 94 checkbox validation bug

OMEGA-94

## What is the reason for this change?
In checkbox list at least one is required and should not show error.

How to reproduce:
- Go to https://www.elvia.no/proff/verktoy-og-skjema/skjemaer/tilknytning-av-storre-anlegg-eller-flytting-av-stromnettet
- Check both checkboxes
- Uncheck one checkbox and see error message

![image-20200903-073253](https://user-images.githubusercontent.com/50363113/93452293-88cd4600-f8d8-11ea-806d-9c74e6fe13bc.png)

Another bug that was found during investigation for this task was that error message was shown below each checkbox:
<img width="278" alt="Screenshot 2020-09-17 at 11 34 17" src="https://user-images.githubusercontent.com/50363113/93453267-bf579080-f8d9-11ea-92e7-b6631e154fec.png">

### What is the expected result?
Checkbox list should not show error message if at least one checkbox is checked. 

<img width="297" alt="Screenshot 2020-09-17 at 11 29 57" src="https://user-images.githubusercontent.com/50363113/93452790-21fc5c80-f8d9-11ea-8342-c2792d50a6a1.png">

Only one error message should be shown below the group of checkboxes.

## How is it solved?
- Added a check in `isRequiredAndDirty` in `FormInputUtils.tsx` that checks if one of the checkboxes are checked. Error is only shown if none of the checkboxes are checked. 
- To ensure that only one error message is shown, I added another condition in the check in `hideError` to not be shown if options are more than one in `InputOptionGroup.tsx`.  

## Any changes/updates that may affect others?

|                       | Changed |
| --------------------- | ------- |
| Unit tests            |        |
| Dependencies          |         |
| Environment variables |         |
| Build scripts         |         |

## How can it be tested and verified?

- [ ] Error is shown if no checkboxes are checked and they are touched
- [ ] Error is not shown if at least one checkbox is checked in a group
